beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_NDT_ELLIPSOID_HPP
+#define BELUGA_ROS_NDT_ELLIPSOID_HPP
+
+#include <beluga_ros/messages.hpp>
+
+#include <Eigen/Core>
+#include <beluga/eigen_compatibility.hpp>
+
+#include <range/v3/view/enumerate.hpp>
+
+#include <beluga/sensor/data/ndt_cell.hpp>
+#include <beluga/sensor/data/sparse_value_grid.hpp>
+
+/**
+ * \file
+ * \brief Utilities for NDT I/O over ROS interfaces.
+ */
+
+namespace beluga_ros {
+
+bool use_mean_covariance(
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>>& eigenSolver,
+    const beluga::NDTCell<3> cell,
+    beluga_ros::msg::Marker& marker);
+
+void use_cell_size(const Eigen::Vector<int, 3>& position, double size, beluga_ros::msg::Marker& marker);
+
+/// Assign an ellipsoid to each cell of a SparseValueGrid. A cube is used instead if the distribution of the
+/// cell is not suitable for the rotation matrix creation.
+/**
+ * \param grid A SparseValueGrid that contains cells representing obstacles.
+ * \return A message with the ellipsoids or cubes.
+ * \param[out] message Markers message to be assigned.
+ * \param[out] cubesGenerated Is set to true if there were problems with covariance matrices from cells.
```

Comment:
style: Use snake_case for variables here and everywhere else.
```suggestion
 * \param[out] cubes_generated Is set to true if there were problems with covariance matrices from cells.
```